Penteract/Assets/Scripts/AIMovement.cpp

```diff
@@ -2,184 +2,65 @@
 
 #include "GameObject.h"
 #include "GameplaySystems.h"
-#include "TesseractEvent.h"
-
-#include "PlayerController.h"
-#include "HUDController.h"
 
 EXPOSE_MEMBERS(AIMovement) {
-    MEMBER(MemberType::GAME_OBJECT_UID, playerUID),
-    MEMBER(MemberType::GAME_OBJECT_UID, canvasUID),
-    MEMBER(MemberType::INT, maxSpeed),
-    MEMBER(MemberType::INT, lifePoints),
-    MEMBER(MemberType::FLOAT, searchRadius),
-    MEMBER(MemberType::FLOAT, meleeRange),
-    MEMBER(MemberType::FLOAT, timeToDie)
-
+    
 };
 
 GENERATE_BODY_IMPL(AIMovement);
 
 void AIMovement::Start() {
-    player = GameplaySystems::GetGameObject(playerUID);
-    if (player) {
-        playerController = GET_SCRIPT(player, PlayerController);
-    }
+    ownerTransform = GetOwner().GetComponent<ComponentTransform>();
     agent = GetOwner().GetComponent<ComponentAgent>();
-    if (agent) {
-        agent->SetMaxSpeed(maxSpeed);
-        agent->SetMaxAcceleration(9999);
-        agent->SetAgentObstacleAvoidance(true);
-        agent->RemoveAgentFromCrowd();
-    }
-    animation = GetOwner().GetComponent<ComponentAnimation>();   
-    parentTransform = GetOwner().GetComponent<ComponentTransform>();
-    GameObject* canvas = GameplaySystems::GetGameObject(canvasUID);
-    if (canvas) {
-        hudControllerScript = GET_SCRIPT(canvas, HUDController);
-    }
 }
 
 void AIMovement::Update() {
-    if (!GetOwner().IsActive()) return;
+    
+}
 
-    if (hitTaken && lifePoints > 0) {
-        lifePoints -= damageRecieved;
-        hitTaken = false;
-    }
+void AIMovement::Seek(AIState state, const float3& newPosition, int speed)
+{
 
-    if (lifePoints <= 0) {
-        if (state == AIState::ATTACK) {
-            animation->SendTrigger("AttackDeath");
-        }
-        else if (state == AIState::IDLE) {
-            animation->SendTrigger("IdleDeath");
-        }
-        else if (state == AIState::RUN) {
-            animation->SendTrigger("RunDeath");
-        }
-        Debug::Log("Death");
-        agent->RemoveAgentFromCrowd();
-        state = AIState::DEATH;
-    }
+    float3 position = ownerTransform->GetGlobalPosition();
```

Comment:
If owner transform is nullptr, this will crash. There's a precondition, that each 3D GameObject has a Component Transform, but just in case check the nullptr.